src/index/vocabulary/SplitVocabulary.h

```diff
@@ -0,0 +1,183 @@
+// Copyright 2025 University of Freiburg
+// Chair of Algorithms and Data Structures
+// Author: Christoph Ullinger <ullingec@cs.uni-freiburg.de>
+
+#ifndef QLEVER_SRC_INDEX_VOCABULARY_SPLITVOCABULARY_H
+#define QLEVER_SRC_INDEX_VOCABULARY_SPLITVOCABULARY_H
+
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <string_view>
+
+#include "global/ValueId.h"
+#include "index/vocabulary/VocabularyTypes.h"
+#include "util/Exception.h"
+#include "util/HashSet.h"
+
+// Func. to decide where a literal goes
+
+// Func. to get filenames for each vocabulary
+
+// using SplitFunctionT = const std::function<bool(std::string_view)>&;
+// using SplitFilenameFunctionT =
+//     const std::function<std::array<std::string, 2>(std::string)>&;
+
+template <class MainVocabulary, class SpecialVocabulary,
+          const auto& SplitFunction, const auto& SplitFilenameFunction>
```

Comment:
Add constraints via the `CPP_template`  macro (e.g. use `ad_utility::InvocableWithExactReturnType` etc. 
And be more explicit with the documentation etc.